pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -335,6 +336,15 @@ private SegmentCompletionFSM(PinotLLCRealtimeSegmentManager segmentManager,
       _winner = "UNKNOWN";
     }
 
+    // Table level setting for Commit timeout
+    private long getMaxSegmentCommitTimeMs(){
+      int maxSegmentCommitTime = _segmentManager.getCommitTimeoutSeconds(_segmentName.getTableName());
```

Comment:
Note that the call to get table config happens multiple times during the commit of a segment. For each table, for each partition, for each segment commit we will get the tableconfig. 

For now, it may be ok, but it may make sense to cache these values and update during a validation manager run (for values that are OK to be updated every hour).  The following things come to mind;
Kafka partition assignment, table-config (commit time), take config (segment flush size). 
May be more.